manifests/linux/redhat.pp

```diff
@@ -62,58 +62,62 @@
     ensure    => $ensure,
     enable    => $enable,
     hasstatus => true,
-    require   => File["/etc/sysconfig/${service_name_v6}"],
+  }
+
+  # Redhat 7 selinux user context for /etc/sysconfig/iptables is set to unconfined_u
+  # Redhat 7 selinux type context for /etc/sysconfig/iptables is set to etc_t
+  case $::selinux {
+    #lint:ignore:quoted_booleans
+    'true',true: {
+      case $::operatingsystemrelease {
+        /^7\..*/: {
+          case $::operatingsystem {
+            'CentOS': {
+              $seluser = 'unconfined_u'
+              $seltype = 'system_conf_t'
+            }
+            default : {
+              $seluser = 'unconfined_u'
+              $seltype = 'etc_t'
+            }
+          }
+        }
+        /^6\..*/:     {
+          $seluser = 'unconfined_u'
+          $seltype = 'system_conf_t'
+        }
+        default:      {
+          $seluser = 'system_u'
+          $seltype = 'system_conf_t'
+        }
+      }
+    }
+    default:     {}
+    #lint:endignore
```

Comment:
The other changes seem fine, but why does this block have to move?

https://github.com/puppetlabs/puppetlabs-firewall/blob/master/lib/puppet/type/firewall.rb#L1531 will also require updating.